node_manager.go

```diff
@@ -0,0 +1,992 @@
+/*
+ * File: node_manager.go
+ *
+ * Description: functions to handle nodes and node creation
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	clientv3 "github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/mvcc/mvccpb"
+	"golang.org/x/net/context"
+)
+
+// The node definition.
+type Node struct {
+	HostName string
+	HostID   string
+}
+
+//! Update the TTL for a given reserved entry.
+/*
+ * @param    LeaseID    id of a given etcd key entry
+ *
+ * @return   none
+ */
+func (inst *EtcdInstance) updateTTL(leaseID clientv3.LeaseID) {
+
+	// Keep the entry alive.
+	_, err := inst.internal.KeepAliveOnce(context.TODO(), leaseID)
+
+	// if an error occurs, dump it to stdout
+	if err != nil {
+		stdlog(err.Error())
+		stdlog("updateTTL() --> could not update TTL")
+	}
+}
+
+//! Function to keep etcd key values alive.
+/*
+ * @param    LeaseGrantResponse    lease response
+ *
+ * @return   none
+ */
+func (inst *EtcdInstance) keepKeyAlive(lease *clientv3.LeaseGrantResponse) {
+
+	// Set a duration time based on the time-to-live, this will be a
+        // sort of 'duration' time.
+	sleep_duration := time.Duration(lease.TTL / 2)
+
+	// infinite loop
+	for {
```

Comment:
there is no way to stop this loop, usually I recommend to use time.Tick instead and check for "context" passed in in select:

```go
select {
   case <-ctx.Done():
      return
    case <- tick.C:
       // time to do things
}
```